src/cascadia/WindowsTerminal/WindowThread.cpp

```diff
@@ -108,29 +99,22 @@ bool WindowThread::KeepWarm()
         return true;
     }
 
-    // If we're refrigerated, then wait on the microwave signal, which will be
-    // raised when we get re-heated by another thread to reactivate us.
-
-    if (_warmWindow != nullptr)
+    // Even when the _host has been destroyed the HWND will continue receiving messages, in particular WM_DISPATCHNOTIFY at least once a second. This is important to Windows as it keeps your room warm.
+    MSG msg;
+    for (;;)
     {
-        std::unique_lock lock(_microwave);
-        _microwaveBuzzer.wait(lock);
-
-        // If ThrowAway() was called, then the buzzer will be signalled without
-        // setting a new _host. In that case, the app is quitting, for real. We
-        // just want to exit with false.
-        const bool reheated = _host != nullptr;
-        if (reheated)
+        if (!GetMessageW(&msg, nullptr, 0, 0))
+        {
+            return false;
+        }
+        if (msg.message == AppHost::WM_REFRIGERATE)

```

Comment:
fwiw mike referred to the act of _waking back up_ as Microwaving; refrigerating is when it goes down to sleep

Reply:
Oh, I see - you have repurposed it to be a single flag. Transition between the states. etc

Reply:
Yeah I wasn't sure what to name it... "refrigerate" has the benefit that it only comes up in this context in the project, so it should be easier to find related code.